src/ocamlutil/pretty.ml

```diff
@@ -255,54 +263,55 @@ type align =
     }
 
 (* We use references to avoid the need to pass data around all the time *)
-let aligns: align list ref =  (* The current stack of active alignment marks,
+let aligns: align list refDLS =  (* The current stack of active alignment marks,
                                  with the top at the head. Never empty.  *)
-  ref [{ gainBreak = 0; isTaken = ref false;
+  refDLS [{ gainBreak = 0; isTaken = ref false;
          deltaFromPrev = ref 0; deltaToNext = ref 0; }]
 
-let topAlignAbsCol = ref 0 (* The absolute column of the top alignment *)
+let topAlignAbsCol = refDLS 0 (* The absolute column of the top alignment *)
 
 let pushAlign (abscol: int) =
-  let topalign = List.hd !aligns in
-  let res =
+  let topalign = List.hd @@ DLS.get aligns in
+  let (res : align) =
     { gainBreak = 0; isTaken = ref false;
       deltaFromPrev = topalign.deltaToNext; (* Share with the previous *)
       deltaToNext = ref 0; (* Allocate a new ref *)} in
-  aligns := res :: !aligns;
-  res.deltaFromPrev := abscol - !topAlignAbsCol;
-  topAlignAbsCol := abscol
+  DLS.set aligns (res :: (DLS.get aligns));
```

Comment:
There's lots of unnecessary parentheses around `DLS,get` now all over this file.